TrafficCapture/captureKafkaOffloader/src/test/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/KafkaCaptureFactoryTest.java

```diff
@@ -96,19 +105,32 @@ private int calculateRecordSize(ProducerRecord<String, byte[]> record, String re
     }
 
     @Test
-    public void testLinearOffloadingIsSuccessful() throws IOException {
+    public void testLinearOffloadingIsSuccessful()
+        throws IOException, InterruptedException, ExecutionException, TimeoutException {
         KafkaCaptureFactory kafkaCaptureFactory =
                 new KafkaCaptureFactory(TestRootKafkaOffloaderContext.noTracking(),
                         TEST_NODE_ID_STRING, mockProducer, 1024*1024);
         var offloader = kafkaCaptureFactory.createOffloader(createCtx());
 
-        List<Callback> recordSentCallbacks = new ArrayList<>(3);
-        when(mockProducer.send(any(), any())).thenAnswer(invocation -> {
+        List<FutureTask<RecordMetadata>> recordSentFutures = new ArrayList<>(3);
+
+        List<CountDownLatch> latches = Arrays.asList(
+            new CountDownLatch(1),
+            new CountDownLatch(1),
+            new CountDownLatch(1)
+        );
+
+        var latchIterator = latches.iterator();
+        when(mockProducer.send(any())).thenAnswer(invocation -> {
```

Comment:
This is a poster child for how mocks create brittle implementation specific tests.  This mock is now only for a variant of send that may or may not be called.  The code that I posted above uses a callback, so this code won't fire.  The test would need to be updated (yet again).